File: Bronze/Bronze_5/Java_files/Code/Receipt_25304.java
import java.util.Scanner;

public class Receipt_25304 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt(), n = sc.nextInt(), sum = 0;
        for (int i = 0; i < n; i++) {
            int a = sc.nextInt(), b = sc.nextInt();
            sum += a * b;
        }
        System.out.println((sum == x) ? "Yes" : "No");
    }
}

File: Bronze/Bronze_5/C_files/Code/Receipt_25304.c
#include <stdio.h>

int main()
{
    int x, n, a, b, i, sum = 0;

    scanf("%d\n%d", &x, &n);

    for (i = 0; i < n; i++) {
        scanf("%d %d", &a, &b);
        sum += a * b;
    }

    printf("%s", (sum == x) ? "Yes" : "No");

    return 0;
}